Test build failure: "coreclr Common Pri1 Test Build AnyOS AnyCPU checked" CI job fails

Apparently starting with https://dev.azure.com/dnceng-public/public/_build/results?buildId=813293&view=results

```
    GCSimulator_127 -> /__w/1/s/artifacts/tests/coreclr/AnyOS.x64.Checked/GC/Scenarios/GCSimulator/GCSimulator_127/GCSimulator_127.dll
/__w/1/s/artifacts/tests/coreclr/obj/AnyOS.x64.Checked/Managed/CoreMangLib/CoreMangLib/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs(618,61): error CS0012: The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Runtime, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. [/__w/1/s/src/tests/CoreMangLib/CoreMangLib.csproj] [/__w/1/s/src/tests/build.proj]
##[error]artifacts/tests/coreclr/obj/AnyOS.x64.Checked/Managed/CoreMangLib/CoreMangLib/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs(618,61): error CS0012: The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Runtime, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. [/__w/1/s/src/tests/CoreMangLib/CoreMangLib.csproj]
    GCSimulator_129 -> /__w/1/s/artifacts/tests/coreclr/AnyOS.x64.Checked/GC/Scenarios/GCSimulator/GCSimulator_129/GCSimulator_129.dll
```

Maybe due to https://github.com/dotnet/runtime/pull/104731? @jkotas @huoyaoyuan

--- a/src/tests/CoreMangLib/system/delegate/miscellaneous/InvokeNull.il
+++ b/src/tests/CoreMangLib/system/delegate/miscellaneous/InvokeNull.il
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 .assembly extern xunit.core {}
-.assembly extern System.Runtime { auto }
+.assembly extern System.Runtime { .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A ) }
 
 .assembly InvokeNull
 {

--- a/src/tests/CoreMangLib/system/delegate/miscellaneous/InvokeNull.ilproj
+++ b/src/tests/CoreMangLib/system/delegate/miscellaneous/InvokeNull.ilproj
@@ -1,12 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk.IL">
   <PropertyGroup>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <CLRTestPriority>1</CLRTestPriority>
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="InvokeNull.il" />
-  </ItemGroup>
-  <ItemGroup>
-    <ProjectReference Include="$(TestSourceDir)Common/CoreCLRTestLibrary/CoreCLRTestLibrary.csproj" />
   </ItemGroup>
 </Project>

